.github/workflows/rust.yml

```diff
@@ -24,62 +24,29 @@ jobs:
       fail-fast: false
       matrix:
         runs-on:
-          - ubuntu-20.04
-          - ubuntu-22.04
-          - macos-11
-          - macos-12
+          # We only need to run unit tests on one OS version.
+          # Saves cache size.
+          - ubuntu-latest
           - macos-13
-          - windows-2019
           - windows-2022
         # TODO: https://github.com/rust-lang/cargo/issues/5220
         include:
-          - runs-on:  ubuntu-20.04
-            packages: -p headless -p gateway
-          - runs-on: ubuntu-22.04
-            packages: -p headless -p gateway
-          - runs-on: macos-11
-            packages: -p connlib-apple
-          - runs-on: macos-12
-            packages: -p connlib-apple
+          - runs-on:  ubuntu-latest
+            packages: -p headless -p gateway -p headless -p connlib-android
           - runs-on: macos-13
             packages: -p connlib-apple
-          - runs-on: windows-2019
-            packages: -p firezone-client-connlib
           - runs-on: windows-2022
             packages: -p firezone-client-connlib
     runs-on: ${{ matrix.runs-on }}
     steps:
       - uses: actions/checkout@v4
-
-      # This implicitly triggers installation of the toolchain in the `rust-toolchain.toml` file.
-      # If we don't do this here, our cache action will compute a cache key based on the Rust version shipped on GitHub's runner which might differ from the one we use.
-      - run: rustup show
-
+      - name: Update toolchain
+        run: rustup show
       - uses: Swatinem/rust-cache@v2
         with:
           workspaces: ./rust
-          key: v2
-          prefix-key: rust-${{ matrix.runs-on }}
+          shared-key: ${{ runner.os }}
```

Comment:
Should shared key still have some prefix? Can it pick up an action cache not related to rust at all?

Reply:
Nah, these are already scoped by default as `v0-rust`